cg/meta/store/base.py

```diff
@@ -1,13 +1,66 @@
 """Base module for building bioinfo workflow bundles for linking in Housekeeper"""
 import datetime as dt
 
+import logging
 from typing import List
 
-from cg.constants import HK_TAGS, MIP_DNA_TAGS, MIP_RNA_TAGS
+from housekeeper.store import models as hk_models
+
+from cg.apps.hk import HousekeeperAPI
+from cg.constants import HK_TAGS, MIP_DNA_TAGS, MIP_RNA_TAGS, MICROSALT_TAGS
+from cg.exc import (
+    AnalysisDuplicationError,
+    PipelineUnknownError,
+    MandatoryFilesMissing,
+    BundleAlreadyAddedError,
+)
+from cg.meta.store.microsalt import add_microbial_analysis
+import cg.meta.store.mip as store_mip
+from cg.store import models, Store
+from cg.store.utils import reset_case_action
+
+from _io import TextIOWrapper
+
+ANALYSIS_TYPE_TAGS = {
+    "wgs": MIP_DNA_TAGS,
+    "wes": MIP_DNA_TAGS,
+    "wts": MIP_RNA_TAGS,
+    "microbial": MICROSALT_TAGS,
+}
+LOG = logging.getLogger(__name__)
+
+
+def gather_files_and_bundle_in_housekeeper(
+    config_stream: TextIOWrapper, hk_api: HousekeeperAPI, status: Store, workflow: str
+) -> models.Analysis:
+    """Function to gather files and bundle in housekeeper"""
+
+    add_analysis = {
+        "microsalt": add_microbial_analysis,
+        "mip": store_mip.add_mip_analysis,
+    }
+    bundle_data = add_analysis[workflow](config_stream)
+
+    results = hk_api.add_bundle(bundle_data)
+    if results is None:
+        raise BundleAlreadyAddedError("bundle already added")
+
+    bundle_obj, version_obj = results
+
+    case_obj = {
+        "microsalt": status.find_family_by_name(bundle_obj.name),
+        "mip": status.family(bundle_obj.name),
```

Comment:
`status.family` uses `internal_id` to get the case object. Microbial orders are identified by ticket number rather than angrybird name (in the deliverables file), and the ticket number was put in the `name` column in PR https://github.com/Clinical-Genomics/cg/pull/775. So it's a slightly different query.